pages/review/[lesson].tsx

```diff
@@ -60,9 +60,9 @@ const Review: React.FC<QueryDataProps<AppData>> = ({ queryData }) => {
     router.push('/curriculum')
     return <LoadingSpinner />
   }
-  const lessonSubmissions: SubmissionData[] = data
+  const lessonSubmissions: Submission[] = data
     ? data.submissions.filter(
-        (submission: SubmissionData) =>
+        (submission: Submission) =>
```

Comment:
Again unnecessary, inferred type

Reply:
It's `any` without this type.

Reply:
Type is `any` because component doesn't use generate query hook. A little personal tip from me, always be cautious of variables with an `any` type, 99% of the time it needs to be fixed. Inferred types can take you long way without needing explicit assertions.

Reply:
The problem is codegen types are not strict enough. Almost every field is optional so it's either a lot of unnecessary null/undefined checks or type assertion again.

Reply:
> it's either a lot of unnecessary null/undefined checks or type assertion again

Yes that's the reason to use typescript, so you don't assume everything is there and get into an undefined value error on production. It's better to do useless null/undefined checks with a safely typed value than having to use an `any` type that you don't know what it is and doesn't work as expected.